src/lightning_app/utilities/proxies.py

```diff
@@ -394,47 +403,51 @@ def run_once(self):
             raise e
         except BaseException as e:
             # 10.2 Send failed delta to the flow.
+            reference_state = deepcopy(self.work.state)
             self.work._calls[call_hash]["statuses"].append(
                 make_status(WorkStageStatus.FAILED, message=str(e), reason=WorkFailureReasons.USER_EXCEPTION)
             )
-            self.delta_queue.put(ComponentDelta(id=self.work_name, delta=Delta(DeepDiff(state, self.work.state))))
+            self.delta_queue.put(
+                ComponentDelta(id=self.work_name, delta=Delta(DeepDiff(reference_state, self.work.state)))
+            )
             self.work.on_exception(e)
             print("########## CAPTURED EXCEPTION ###########")
             print(traceback.print_exc())
             print("########## CAPTURED EXCEPTION ###########")
             return
 
-        # 14. Copy all artifacts to the shared storage so other Works can access them while this Work gets scaled down
-        persist_artifacts(work=self.work)
-
-        # 15. Destroy the state observer.
+        # 13. Destroy the state observer.
         self.state_observer.join(0)
         self.state_observer = None
 
+        # 14. Copy all artifacts to the shared storage so other Works can access them while this Work gets scaled down
+        persist_artifacts(work=self.work)
+
         # 15. An asynchronous work shouldn't return a return value.
         if ret is not None:
             raise RuntimeError(
                 f"Your work {self.work} shouldn't have a return value. Found {ret}."
                 "HINT: Use the Payload API instead."
             )
 
-        # 16. DeepCopy the state and send the latest delta to the flow.
+        # 17. DeepCopy the state and send the latest delta to the flow.
         # use the latest state as we have already sent delta
         # during its execution.
         # inform the task has completed
-        state = deepcopy(self.work.state)
+        reference_state = deepcopy(self.work.state)
         self.work._calls[call_hash]["statuses"].append(make_status(WorkStageStatus.SUCCEEDED))
         self.work._calls[call_hash]["ret"] = ret
-        self.delta_queue.put(ComponentDelta(id=self.work_name, delta=Delta(DeepDiff(state, self.work.state))))
+        self.delta_queue.put(ComponentDelta(id=self.work_name, delta=Delta(DeepDiff(reference_state, self.work.state))))
```

Comment:
I'm still failing to see why delta here would be different from zero, since we just deepcopy'ed and _calls is not part of the circulating state at this point.

Reply:
`_calls` is part of the circulating state, but only the section associated with the current request call hash.